assets/components/src/plugin-settings/SettingsSection.js

```diff
@@ -0,0 +1,101 @@
+/**
+ * Ads Settings Section.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { Fragment } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { ActionCard, Grid, Button, TextControl, CheckboxControl, SelectControl } from '../';
+
+const SettingsSection = ( { settings, disabled, onChange, onUpdate } ) => {
+	const section = settings.find( setting => ! setting.key || setting.key === 'active' );
+	if ( ! section ) {
+		return null;
+	}
+	const activation = settings.find( setting => setting.key === 'active' );
+	const fields = settings.filter( setting => setting.key && setting.key !== 'active' );
```

Comment:
The logic with `'active'` value of `setting.key` is pretty confusing. I would not readily understand how does `settings` prop shape look like (an array with objects with `key` key which must be `'active'` in order for the setting to be displayed?) if I wanted to use this component.
Off the top of my head, looking at the UI, I'd imagine `SettingsSection` taking following props (aside from read/write handlers):
- active state
- title
- subheader
- a list of fields


<img width="850" alt="image" src="https://user-images.githubusercontent.com/7383192/140904884-f51d22c4-0563-4816-bed1-9a4d2e237d72.png">

I'd suggest moving the settings-wrangling logic to the parent component, which deals with the [proposed API](https://github.com/Automattic/newspack-ads/pull/213).

Reply:
Much agreed! Done in fb2b80d04188e0287f7e8e31e104de485ecd27e9

Reply:
Note that the sections are not necessarily modules with activation. Therefore, the `active` property can be `null` and the component should be able to handle it as well.

Reply:
Update: ce9d1ccad8e370d5d0a6abdae017d9ee2cdf991f handles a remaining `fields` logic left on the abstract component.